src/CBLCollection_Internal.hh

```diff
@@ -147,21 +147,7 @@ public:
             C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "centroids must be >= 1 and <= 64000.");
         }
         
-        // Use default minTrainingSize if minTrainingSize is not specified:
-        if (config.minTrainingSize == 0) {
-            config.minTrainingSize = 25 * config.centroids;
-        }
-        
-        // Use default maxTrainingSize if maxTrainingSize is not specified:
-        if (config.maxTrainingSize == 0) {
-            config.maxTrainingSize = 250 * config.centroids;
-        }
-        
-        if (config.minTrainingSize < 1) {
-            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "minTrainingSize must be > 1.");
-        } else if (config.maxTrainingSize < 1) {
-            C4Error::raise(LiteCoreDomain, kC4ErrorInvalidParameter, "maxTrainingSize must be > 1.");
-        } else if (config.minTrainingSize > config.maxTrainingSize) {
+        if (config.minTrainingSize > 0 && config.maxTrainingSize > 0 && config.minTrainingSize > config.maxTrainingSize) {
```

Comment:
This condition is needlessly long.  Why not just the last part?

Reply:
because config.minTrainingSize and config.maxTrainingSize can be zero which means the real value will be calculated in core.

Reply:
Oh....I guess I need to fix that in my code then.  I never thought of just setting one or the other 😩